File: CarCareHub_/CarCareHub.Services/Database/ChatAutoservisKlijent.cs
using System;

namespace CarCareHub.Services.Database
{
    public class ChatAutoservisKlijent
    {
        public int Id { get; set; }
        public int KlijentId { get; set; }
        public int AutoservisId { get; set; }
        public string Poruka { get; set; }
        public bool PoslanoOdKlijenta { get; set; } 
        public DateTime VrijemeSlanja { get; set; }
        public bool Vidljivo { get; set; }
        public virtual Klijent Klijent { get; set; }
        public virtual Autoservis Autoservis { get; set; }
    }
}

File: CarCareHub_/UI_/flutter_desktop/lib/models/chatAutoservisKlijent.dart
import 'package:flutter_mobile/models/autoservis.dart';
import 'package:flutter_mobile/models/klijent.dart';
import 'package:json_annotation/json_annotation.dart';
part 'chatAutoservisKlijent.g.dart';

@JsonSerializable()
class chatAutoservisKlijent {
  int id;
  int klijentId;
  String klijentIme;
  int autoservisId;
  String autoservisNaziv;
  String? poruka;
  bool poslanoOdKlijenta;
  DateTime vrijemeSlanja;


  chatAutoservisKlijent(this.id,  this.klijentId,this.autoservisId, this.autoservisNaziv, this.klijentIme, 
  this.poruka,this.poslanoOdKlijenta,this.vrijemeSlanja);

  
  factory chatAutoservisKlijent.fromJson(Map<String,dynamic> json) => _$chatAutoservisKlijentFromJson(json);


  Map<String,dynamic> toJson() => _$chatAutoservisKlijentToJson(this);
}
